storage/innobase/xtrabackup/src/keyring_plugins.cc

```diff
@@ -200,23 +205,23 @@ static bool xb_fetch_key(char *key_name, char *key) {
 @param[out]	key_name	transition key name
 @return false if fetch has failed. */
 static bool xb_create_transition_key(char *key_name, char *key) {
-  byte rand32[32];
-  char rand64[45];
+  byte rand32[20];
```

Comment:
shouldn't this be 32 bytes? it seems the intention is create 64byte random key/prefix?

Reply:
How will this work vault v1? How will PXB read keys which are in v1 vault version?

Reply:
This is not a problem.
This is a Key=>value secret we store on vault. It could be anything. we then store it on xtrabackup_keys:
```
strings /tmp/backup/xtrabackup_keys
KEYSV01W=
XBKey-c70f89d3-1af1-11ec-9b69-d45d64347a19-tUTpWs/F+vwY1VJTgZdeeOKFEsA=
}~y
SrYb
}~y
SrYb
```
Here `XBKey-c70f89d3-1af1-11ec-9b69-d45d64347a19-tUTpWs/F+vwY1VJTgZdeeOKFEsA=` is the the key, and the last part `tUTpWs/F+vwY1VJTgZdeeOKFEsA=` is what we generated. At prepare we will read this key from xtrabackup_keys file and fetch it from keyring (vault or file). This is just stamp we add on each backup to avoid colisions. We are still generating an 30 ascii char len sufix key, which will be more than enough for backups&increments to not colide.

For example, the server implements it by using an incremental integer as suffix.

Reply:
ok, the readers don't insist the key to be of specific len and pattern. Got it.